projects/noloan2/app/src/main/java/hasadna/noloan2/SplashPermissionActivity.java

```diff
@@ -0,0 +1,77 @@
+package hasadna.noloan2;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+import noloan.R;
+
+// Base on http://pcessflight.com/smart-android-splash-screen-grabbing-permissions/
+public class SplashPermissionActivity extends AppCompatActivity {
+  
+  private static final int PERMISSION_REQUEST_CODE = 123;
+  final String[] permissionsNeeded = new String[]{Manifest.permission.READ_SMS};
+  
+  long splashTime = 1000;
+  
+  long startTime;
+  
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_splash_permission);
+    
+    startTime = System.currentTimeMillis();
+    
+    if (Build.VERSION.SDK_INT >= 23) {
+      checkPermissions();
+    } else {
+      startNextActivity();
+    }
+  }
+  
+  private void checkPermissions() {
+    String[] ungrantedPermissions = permissionsStillNeeded();
+    if (ungrantedPermissions.length == 0) {
+      startNextActivity();
+    } else {
+      Toast.makeText(this, "Permission for reading sms required", Toast.LENGTH_LONG).show();
+      requestPermissions(ungrantedPermissions, PERMISSION_REQUEST_CODE);
+    }
+  }
+  
+  private String[] permissionsStillNeeded() {
```

Comment:
Lists are generally nicer than arrays :)
I think it's nicer to return a list `List<String>` and then do `Strings.toArray(list)` in `checkPermissions()`.